web/modules/custom/contact_information/contact_information.module

```diff
@@ -112,29 +112,50 @@ function contact_information_entity_auto_label(EntityInterface $entity) {
  * Implements hook_ENTITY_TYPE_access().
  */
 function contact_information_rdf_entity_access(EntityInterface $entity, $operation, AccountInterface $account) {
-  // Access being checked for a contact information.
-  if ($entity->getEntityTypeId() != 'rdf_entity') {
+  if ($entity->bundle() !== 'contact_information') {
     return AccessResult::neutral();
   }
-  if ($entity->bundle() != 'contact_information') {
-    return AccessResult::neutral();
-  }
-  // Only revoke access for anonymous users.
-  if ($account->isAuthenticated()) {
-    return AccessResult::neutral();
+
+  // Inline entity form module is using the 'update' as a name for the edit
+  // operation as it is with all entities apart from nodes that retained the
+  // 'edit' permission for legacy reasons. However, rdf entities also use this
+  // naming convention. Rename 'update' to 'edit' when inline_entity_form checks
+  // for the update permission.
+  if ($operation === 'update') {
+    $operation = 'edit';
   }
-  // Get entity from route.
-  $route = \Drupal::routeMatch();
-  $route_name = $route->getRouteName();
-  if ($route_name != 'entity.rdf_entity.canonical') {
-    return AccessResult::neutral();
+
+  // For anonymous users, deny access to the canonical route.
+  if ($account->isAnonymous()) {
+    $route_name = \Drupal::routeMatch()->getRouteName();
+    if ($route_name === 'entity.rdf_entity.canonical') {
+      return AccessResult::forbidden('Anonymous users should not see contact information on the canonical route.');
+    }
   }
-  $route_entity = $route->getParameter('rdf_entity');
-  if (empty($route_entity)) {
-    return AccessResult::neutral();
+  elseif ($operation == 'edit') {
+    $allowed_transitions = \Drupal::service('joinup_core.workflow.helper')->getAvailableTargetStates($entity, $account);
+    return empty($allowed_transitions) ? AccessResult::forbidden() : AccessResult::allowed();
   }
-  if ($entity->id() == $route_entity->id()) {
-    return AccessResult::forbidden('Anonymous users should not see contact information on the canonical route.');
+  // Use an elseif to avoid checking memberships for anonymous users.
+  elseif ($operation === 'delete') {
+    // The contact information is not a group content rather than it is simply
+    // referenced by a collection or a solution.
```

Comment:
IIRC contact information can be referenced by multiple collections and solutions. This documentation seems to hint that it can only be referenced by a single group.